eo-maven-plugin/src/main/java/org/eolang/maven/OyRemote.java

```diff
@@ -34,34 +38,41 @@
  *
  * @since 0.1
  */
+@SuppressWarnings({"PMD.AvoidDuplicateLiterals",
+    "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"})
 public final class OyRemote implements Objectionary {
 
+    /**
+     * Tag.
+     */
+    private final String tag;
+
     /**
      * The address template.
      */
-    private final String template;
+    private String template;
 
     /**
-     * Ctor.
+     * Constructor.
+     * @param tag Tag
      */
-    public OyRemote() {
-        this("master");
+    public OyRemote(final String tag) {
+        this.tag = tag;
     }
 
     /**
-     * Ctor.
-     * @param hash The GitHub hash
-     * @todo #490:30m Resolve abbreviated hash to a proper hash.
-     *  In order to avoid collisions resolve hash
-     *  (or branch) to a complete sha-256 hash of the commit.
-     *  Use only sha-256 hashes as a caching criteria.
+     * Init class.
+     * @return OyRemote
+     * @throws IOException if fails
      */
-    public OyRemote(final String hash) {
+    public OyRemote resolve() throws IOException {
```

Comment:
@yasamprom this is not a good idea. We should try to keep our objects immutable. I suggest you create a new class just for tag-to-hash resolving. Maybe `HashOfTag` could be its name.